packages/payment-processor/src/payment/index.ts

```diff
@@ -116,6 +120,37 @@ export async function payRequest(
   }
 }
 
+/**
+ * Encode the transactions associated to a request
+ * @param request the request to pay.
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param from The address which will send the transaction.
+ * @param options encoding options
+ * @returns
+ */
+export async function encodeRequest(
```

Comment:
see my other comment regarding the responsability of the encoding function

name is too generic

```suggestion
export function encodeRequestPayment(
```

Reply:
I modified this one to `encodeRequestApprovalAndPayment`. The specific methods for approval or payment encoding can be found in `encoder-approval` and `encoder-payment`.